File: examples/octave_modular/custom_kernel.m
init_shogun;

C=1;
dim=7;

lab=sign(2*rand(1,dim) - 1);
data=rand(dim, dim);
symdata=data+data';

kernel=CustomKernel();
kernel.set_full_kernel_matrix_from_full(data);
labels=Labels(lab);
svm=LibSVM(C, kernel, labels);
svm.train();
out=svm.classify().get_labels();

File: examples/python_modular/custom_kernel.py
from numpy import *
from numpy.random import rand
from shogun.Features import RealFeatures, Labels
from shogun.Kernel import CustomKernel
from shogun.Classifier import LibSVM

C=1
dim=7

lab=sign(2*rand(dim) - 1)
data=rand(dim, dim)
symdata=data+data.T

kernel=CustomKernel()
kernel.set_full_kernel_matrix_from_full(data)
labels=Labels(lab)
svm=LibSVM(C, kernel, labels)
svm.train()
out=svm.classify().get_labels()
